test/integration/coscheduling_test.go

```diff
@@ -0,0 +1,286 @@
+	// Expect Pods are popped in the Coscheduling class order.
+	for i := min - 2; i >= 0; i-- {
+		podInfo := testCtx.Scheduler.NextPod()
+		pgName2, _ := coscheduling.GetPodGroupLabels(podInfo.Pod)
+		if pgName1 != pgName2 {
+			t.Errorf("PodGroup %q/%q is popped out as expected %v.", pgName2, podInfo.Pod.Name, pgName1)
+		} else {
+			t.Logf("PodGroup %q/%q is popped out as expected.", pgName2, podInfo.Pod.Name)
+		}
+	}
+
+	// Get the second podGroup
+	podInfo = testCtx.Scheduler.NextPod()
+	pgName1, min = coscheduling.GetPodGroupLabels(podInfo.Pod)
+	// Expect Pods are popped in the Coscheduling class order.
+	for i := min - 2; i >= 0; i-- {
+		podInfo := testCtx.Scheduler.NextPod()
+		pgName2, _ := coscheduling.GetPodGroupLabels(podInfo.Pod)
+		if pgName1 != pgName2 {
+			t.Errorf("PodGroup %q/%q is popped out as expected %v.", pgName2, podInfo.Pod.Name, pgName1)
+		} else {
+			t.Logf("PodGroup %q/%q is popped out as expected.", pgName2, podInfo.Pod.Name)
+		}
+	}
+}
+
+func TestCoschedulingPlugin(t *testing.T) {
+	registry := framework.Registry{coscheduling.Name: coscheduling.New}
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			QueueSort: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+			PreFilter: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+			Permit: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+			Unreserve: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+		},
+	}
+	testCtx := util.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestMaster(t, "sched-coscheduling", nil),
+		false,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns := testCtx.ClientSet, testCtx.NS.Name
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name("fake-node").Label("node", nodeName).Obj()
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(300, resource.DecimalSI),
+	}
+	node, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	// Create 6 Pods belongs to two podGroup.
+	// pg1 has three pods, and the resources of the cluster can't meet the min of pg1.
+	// pg2 has three pods, but the resources of the cluster can meet the min of pg2.
+	var pods []*v1.Pod
+	pgNames1 := []string{"pod1-1", "pod1-2", "pod1-3"}
+	pgNames2 := []string{"pod2-1", "pod2-2", "pod2-3"}
```

Comment:
I'd like to see more test cases:

1. 2 PodGroups with different pod number - like one has 2, the other has 5.
1. regular Pods are involved
1. 2 PodGroups with different priority (preemption may be triggered) 
1. (optional) any corner case needs to involve more than 2 PodGroups?

And the Pods can be composed in an **arbitrary** order.